### ISSUE TITLE
scheduler crash when use numa-aware plugin in k8s version 1.21

### ISSUE THREAD

**Issue by Jason-Liu-Dream** [volcano-sh/volcano#1845]
**What happened**:
We need use numa-aware plugin, and just build environment refer to NUMA Aware guide. But when we deploy vcjob yaml, the scheduler crash and can not restart.
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
```
apiVersion: batch.volcano.sh/v1alpha1
kind: Job
metadata:
  name: volcano-test
  namespace: volcano-audio
spec:
  maxRetry: 3
  minAvailable: 10
  queue: audio
  policies:
  - event: PodFailed
    action: RestartJob
  tasks:
    - replicas: 10
      name: audio
      topologyPolicy: best-effort
      template:
        metadata:
          name: audio
          annotations:
            iqiyi.com/cpu-share: "4"
        spec:
          schedulerName: volcano
          containers:
          - command:
            - python
            - xxx
            env:
            - name: POD_NAME
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: metadata.name
            - name: NODE_NAME
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: spec.nodeName
            image: xxx
            imagePullPolicy: IfNotPresent
            name: main-container
            resources:
              limits:
                cpu: 4
                ephemeral-storage: 100Gi
                memory: 8Gi
              requests:
                cpu: 4
                ephemeral-storage: 100Mi
                memory: 8Gi
            securityContext:
              privileged: true
            terminationMessagePath: /dev/termination-log
            terminationMessagePolicy: File
            volumeMounts:
            - mountPath: /data/tmp
              name: transcode-data
            - mountPath: /etc/qcs
              name: qcs
            - mountPath: /resources
              name: resources-cache
            - mountPath: /mnt
              name: mnt
          dnsPolicy: ClusterFirst
          hostNetwork: true
          restartPolicy: Never
          securityContext: {}
          shareProcessNamespace: true
          terminationGracePeriodSeconds: 30
          volumes:
          - emptyDir: {}
            name: transcode-data
          - hostPath:
              path: /etc/qcs
              type: ""
            name: qcs
          - hostPath:
              path: /data/resources
              type: ""
            name: resources-cache
          - hostPath:
              path: /mnt
              type: ""
            name: mnt
  ttlSecondsAfterFinished: 180
```
**Anything else we need to know?**:
![version](https://user-images.githubusercontent.com/12228960/142824884-4ef40b7d-61d5-46f8-a57c-403d16a95e4c.png)

scheduler crash log

[volcano_numa.log](https://github.com/volcano-sh/volcano/files/7579378/volcano_numa.log)

scheduler restart log
```
2021/11/22 07:57:22 maxprocs: Leaving GOMAXPROCS=4: CPU quota undefined
W1122 07:57:22.097939       1 client_config.go:608] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I1122 07:57:22.161864       1 event_handlers.go:526] Add Queue(default) into cache, spec(v1beta1.QueueSpec{Weight:1, Capability:v1.ResourceList(nil), Reclaimable:(*bool)(0xc0004a8b3d), ExtendClusters:[]v1beta1.Cluster(nil), Guarantee:v1beta1.Guarantee{Resource:v1.ResourceList(nil)}})
I1122 07:57:22.161926       1 event_handlers.go:526] Add Queue(audio) into cache, spec(v1beta1.QueueSpec{Weight:1, Capability:v1.ResourceList(nil), Reclaimable:(*bool)(0xc0004a8c1c), ExtendClusters:[]v1beta1.Cluster(nil), Guarantee:v1beta1.Guarantee{Resource:v1.ResourceList(nil)}})
W1122 07:57:22.163257       1 warnings.go:67] scheduling.k8s.io/v1beta1 PriorityClass is deprecated in v1.14+, unavailable in v1.22+; use scheduling.k8s.io/v1 PriorityClass
W1122 07:57:22.167579       1 warnings.go:67] scheduling.k8s.io/v1beta1 PriorityClass is deprecated in v1.14+, unavailable in v1.22+; use scheduling.k8s.io/v1 PriorityClass
E1122 07:57:22.173498       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 48 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x18cf420, 0x28b02d0)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x86
panic(0x18cf420, 0x28b02d0)
	/opt/hostedtoolcache/go/1.16.8/x64/src/runtime/panic.go:965 +0x1b9
volcano.sh/volcano/pkg/scheduler/api.(*NodeInfo).setNodeState(0xc0006a40b0, 0x0)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/pkg/scheduler/api/node_info.go:236 +0x3d9
volcano.sh/volcano/pkg/scheduler/api.NewNodeInfo(0x0, 0xc00027fb60)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/pkg/scheduler/api/node_info.go:114 +0x30e
volcano.sh/volcano/pkg/scheduler/cache.(*SchedulerCache).addNumaInfo(0xc000243000, 0xc000011500, 0x1a9170e, 0x2)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/pkg/scheduler/cache/event_handlers.go:809 +0x6d2
volcano.sh/volcano/pkg/scheduler/cache.(*SchedulerCache).AddNumaInfoV1alpha1(0xc000243000, 0x1ad6ce0, 0xc000011500)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/pkg/scheduler/cache/event_handlers.go:855 +0x91
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/client-go/tools/cache/controller.go:227
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/client-go/tools/cache/shared_informer.go:777 +0xc2
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0005b9f60)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0003f9f60, 0x1d2a000, 0xc000160300, 0x1894901, 0xc000256240)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0005b9f60, 0x3b9aca00, 0x0, 0x1bd5701, 0xc000256240)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000498080)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/client-go/tools/cache/shared_informer.go:771 +0x95
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc0002194f0, 0xc00017e930)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:73 +0x51
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x65
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x16a2099]

goroutine 48 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x109
panic(0x18cf420, 0x28b02d0)
	/opt/hostedtoolcache/go/1.16.8/x64/src/runtime/panic.go:965 +0x1b9
volcano.sh/volcano/pkg/scheduler/api.(*NodeInfo).setNodeState(0xc0006a40b0, 0x0)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/pkg/scheduler/api/node_info.go:236 +0x3d9
volcano.sh/volcano/pkg/scheduler/api.NewNodeInfo(0x0, 0xc00027fb60)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/pkg/scheduler/api/node_info.go:114 +0x30e
volcano.sh/volcano/pkg/scheduler/cache.(*SchedulerCache).addNumaInfo(0xc000243000, 0xc000011500, 0x1a9170e, 0x2)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/pkg/scheduler/cache/event_handlers.go:809 +0x6d2
volcano.sh/volcano/pkg/scheduler/cache.(*SchedulerCache).AddNumaInfoV1alpha1(0xc000243000, 0x1ad6ce0, 0xc000011500)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/pkg/scheduler/cache/event_handlers.go:855 +0x91
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/client-go/tools/cache/controller.go:227
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/client-go/tools/cache/shared_informer.go:777 +0xc2
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0005b9f60)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000843f60, 0x1d2a000, 0xc000160300, 0x1894901, 0xc000256240)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0005b9f60, 0x3b9aca00, 0x0, 0x1bd5701, 0xc000256240)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000498080)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/client-go/tools/cache/shared_informer.go:771 +0x95
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc0002194f0, 0xc00017e930)
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:73 +0x51
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/home/runner/work/volcano/volcano/src/github.com/volcano-sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x65
```
**Environment**:
- Volcano Version: release-1.4.0
- Kubernetes version (use `kubectl version`): v1.21.0
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by huone1** [comment]
it is fixed in PR #1800

**Comment by k82cn** [comment]
> it is fixed in PR #1800

Which release are we going to include this fix?

**Comment by Jason-Liu-Dream** [comment]
> it is fixed in PR #1800

We tried it，scheduler can restart when it crash. But first time to commit vcjob，it crash again.  “fatal error: concurrent map writes” maybe the real reason.  refer to above log named volcano_numa.log

**Comment by Jason-Liu-Dream** [comment]
> > it is fixed in PR #1800
> 
> We tried it，scheduler can restart when it crash. But first time to commit vcjob，it crash again. “fatal error: concurrent map writes” maybe the real reason. refer to above log named volcano_numa.log

Try to fixed this bug： https://github.com/volcano-sh/volcano/pull/1847

**Comment by k82cn** [comment]
Great! Let's cherry pick those two issues to 1.4 release :)
